app/code/Magento/Catalog/Controller/Adminhtml/Category/Add.php

```diff
@@ -22,32 +25,40 @@ class Add extends \Magento\Catalog\Controller\Adminhtml\Category implements Http
      */
     protected $resultForwardFactory;
 
+    /**
+     * @var \Magento\Framework\View\Result\PageFactory
+     */
+    protected $resultPageFactory;
+
     /**
      * Add category constructor
      *
      * @param \Magento\Backend\App\Action\Context $context
      * @param \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory
+     * @param \Magento\Framework\View\Result\PageFactory $resultPageFactory
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
-        \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory
+        \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory,
+        \Magento\Framework\View\Result\PageFactory $resultPageFactory
     ) {
         parent::__construct($context);
         $this->resultForwardFactory = $resultForwardFactory;
+        $this->resultPageFactory = $resultPageFactory;
     }
 
     /**
      * Add new category form
      *
-     * @return \Magento\Backend\Model\View\Result\Forward
+     * @return Page | Redirect | Json
```

Comment:
This should return the ResultInterface: https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/Controller/ResultInterface.php which is what is returned by the ResultFactory https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/Controller/ResultFactory.php#L81